### PR TITLE
Remove deprecated code

### PR DESCRIPTION
**Pull request by adekusar-drl**
### Summary
This PR removes the code that was deprecated in 0.5.0
- completes removal of `user_parameters` in `QuantumKernel`
- removes deprecated `quantum_instance` from `VQC` and `VQR`.

### CODE REVIEW

**Review by woodsp-ibm — COMMENTED** [review]
I note `algorithms\serializable_model.py` mentions quantum instance in the docstring around save(). Since save() is inherited by VQC and now shows up in the class maybe that text can get updated to remove such reference.

File: qiskit_machine_learning/kernels/quantum_kernel.py [hunk 1]
```diff
@@ -313,55 +309,6 @@ def get_unbound_training_parameters(self) -> List[Parameter]:
...
-        return self.training_parameters
-
-    @user_parameters.setter
-    @deprecate_property("0.5.0", new_name="training_parameters")
-    def user_parameters(self, training_params: ParameterVector | Sequence[Parameter]) -> None:
-        """[Deprecated property setter]Set the training parameters"""
-        self.training_parameters = training_params
-
-    @deprecate_method("0.5.0", new_name="assign_training_parameters")
-    def assign_user_parameters(
```
[the remark this answers is not in the document]
Comment (adekusar-drl) [reply qiskit_machine_learning/kernels/quantum_kernel.py:329]: Updated the reno file

File: qiskit_machine_learning/algorithms/regressors/vqr.py [hunk 2]
```diff
@@ -99,67 +89,35 @@ def __init__(
             ValueError: if the type of the observable is not compatible with ``quantum_instance`` or
                 ``estimator``.
         """
-        # needed for mypy
-        if quantum_instance is not None and estimator is None:
-            warn_deprecated(
-                "0.5.0", DeprecatedType.ARGUMENT, old_name="quantum_instance", new_name="estimator"
+        if observable is not None and not isinstance(observable, (BaseOperator, PauliSumOp)):
```
[the remark this answers is not in the document]
Comment (adekusar-drl) [reply qiskit_machine_learning/algorithms/regressors/vqr.py:85]: Removed references to quantum instance and different operator types.

File: qiskit_machine_learning/algorithms/classifiers/vqc.py [hunk 3]
```diff
@@ -80,22 +76,14 @@ def __init__(
...
             initial_point: Initial point for the optimizer to start from.
             callback: a reference to a user's callback function that has two parameters and
                 returns ``None``. The callback can access intermediate data during training.
                 On each iteration an optimizer invokes the callback and passes current weights
                 as an array and a computed value as a float of the objective function being
                 optimized. This allows to track how well optimization / training process is going on.
-            sampler: If a sampler instance is sent, the underlying QNN will be of type
-                :class:`~qiskit_machine_learning.neural_networks.SamplerQNN`, and the sampler
-                primitive will be used to compute the neural network's results.
+            sampler: a Sampler primitive instance to be used by the underlying
```
[the remark this answers is not in the document]
Comment (adekusar-drl) [reply qiskit_machine_learning/algorithms/classifiers/vqc.py]: Updated the docstring for `sampler` and `estimator` in `VQC` and `VQR` respectively.

**Comment by adekusar-drl** [comment]
> I note `algorithms\serializable_model.py` mentions quantum instance in the docstring around save(). Since save() is inherited by VQC and now shows up in the class maybe that text can get updated to remove such reference.

Updated the docstrings

File: qiskit_machine_learning/algorithms/regressors/vqr.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (woodsp-ibm) [reply qiskit_machine_learning/algorithms/regressors/vqr.py:85]: You still kept the types in the pbservable typehint then that were only valid for QI i.e. QuantumCircuit and OperatorBase. Relates to the part of the above comment about the error being raised here if its one of these.

>observable: QuantumCircuit | OperatorBase | BaseOperator | PauliSumOp | None = None,

[the remark this answers is not in the document]
Comment (adekusar-drl) [reply qiskit_machine_learning/algorithms/regressors/vqr.py:85]: Fixed the typehint, thanks for spotting.

[the remark this answers is not in the document]
Comment (woodsp-ibm) [reply qiskit_machine_learning/algorithms/regressors/vqr.py:85]: What about the type check/ValueError that was added - last sentence of the first comment above. You still want to go with that?

[the remark this answers is not in the document]
Comment (adekusar-drl) [reply qiskit_machine_learning/algorithms/regressors/vqr.py:84]: Ah, you are totally right, fixed now. I hope :)

**Review by woodsp-ibm — APPROVED** [review]

**Pull request merged**